### PR TITLE
Now constructors have size argument

### PR DESCRIPTION
**Pull request by Raidenkyu**
Just like suggested in [#18328](https://github.com/godotengine/godot/issues/18328), now both Pool*Array and Array have constructors for a predetermined size.

For example, if you want an Array with size 3, you only need to to declare your Array as: var arr = Array(3).
The variable arr will be an Array with size 3. The same logic applies to Pool*Arrays containers.

Also if the user wants to initialize all values with a specific value, there is also a version of both constructors with 2 arguments where the first one is the size of the container and the the second argument is the value of the all elements of the container when initialized. Exemple: var pool = PoolVector3Array(3,Vector3(1,2,3)). The variable pool will be an PoolVector3Array with size 3 and all if its elements are vectors with x=1,y=2,z=3.

## Changes
To implement the new constructors, they were added with **_VariantCall::add_constructor** in **variant_call.pp.** To accept arguments of type int, the method **Variant::can_convert** in **variant.cpp** was modified. The next modified method was **Variant::construct**, where in case of 1 argument of the constructor and the condition **(!p_strict || Variant::can_convert(p_args[0]->type, p_type))** is true, instead calling the defaults constructors of Array and Pool*Array, calls the new static methods: **static  Array Array::sizeInit(int size)** and **static Pool Vector<T> PoolVector<T> ::sizeInit(int size)**. Here is the body of both new methods respectively:

In array.cpp:
<pre><code>
Array Array::sizeInit(int size){
	Array arr;
	arr.resize(size);
	return arr;
}
</pre></code>

In dvector.h
<pre><code>
static PoolVector<T> sizeInit(int size){
	PoolVector<T> pool;
	pool.resize(size);
	return pool;
}
</pre></code>

## To Do
Update the documentation, to include the new methods created.

### CODE REVIEW

File: core/array.cpp [hunk 1]
```diff
@@ -413,7 +413,11 @@ Array::Array() {
 	_p->refcount.init();
 }
 
+<<<<<<< HEAD
```
Comment (Zireael07) [inline core/array.cpp]: Failed merge markers!

Comment (Raidenkyu) [reply core/array.cpp]: In the last commit, every merge conflict is resolved and no markers were left.

**Comment by Zylann** [comment]
Collides with #23830 ?

**Comment by Raidenkyu** [comment]
> Collides with [#23830](https://github.com/godotengine/godot/pull/23830) ?

Probably yes. The development of this solution started before [#23830](https://github.com/godotengine/godot/pull/23830)  was published. I don't recommend using both.

File: core/array.cpp [hunk 2]
```diff
@@ -412,6 +412,13 @@ Array::Array() {
 	_p = memnew(ArrayPrivate);
 	_p->refcount.init();
 }
+
+Array Array::sizeInit(int size) {
```
Comment (aaronfranke) [inline core/array.cpp]: I don't see why this should be camelCase when everything else uses snake_case. This should probably be changed to `init_size` (assuming that the proposal is accepted)

File: core/dvector.h [hunk 3]
```diff
@@ -462,6 +462,19 @@ class PoolVector {
...
 	void operator=(const PoolVector &p_dvector) { _reference(p_dvector); }
 	PoolVector() { alloc = NULL; }
+
+	static PoolVector<T> sizeInit(int size) {
+		PoolVector<T> pool;
+		pool.resize(size);
+		return pool;
+	}
+
+	void setValues(const T &value) {
```
Comment (aaronfranke) [inline core/dvector.h]: Same thing here.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: core/array.cpp (hunk 2 above)
Comment (akien-mga) [reply core/array.cpp]: And method paramaters should start with `p_`, so it should be `Array::init_size(int p_size)`.

File: core/dvector.h (hunk 3 above)
Comment (akien-mga) [reply core/dvector.h]: And `p_values` for the parameter.

**Comment by aaronfranke** [comment]
Is there an update on this? Also, the title should contain "array".

**Comment by Raidenkyu** [comment]
> Is there an update on this? Also, the title should contain "array".

Nope. I didn't work on this anymore

**Pull request closed without merging**